### ISSUE TITLE
Some questions on how to use the code from Causal Agent

### ISSUE THREAD

**Issue by jbdatascience** [Kairong-Han/Causal_Agent#1]
Very interesting application of LLMs in the domain of Causal Inference!

I would like to replicate your results using your code, (not using GPT4 as LLM, but one or more of the open LLMs from GROQ, see for example https://github.com/groq/groq-api-cookbook).

My questions : 

- Can you tell me the exact steps I have to do using your code? It is not clear to me what to run in which order. Especially which output files have to be generated by which code in which order to do this.

- Also I would like to know the process steps at a high level. Can you summarize them for me?

- Where in the code are the numerical data files generated? 

Is that:
https://github.com/Kairong-Han/Causal_Agent/blob/main/SCORE-main/demo_SCORE.py
?

- Also in your paper you mention “we generated a test dataset of 1.3K using ChatGPT-3.5 for these four levels of issue …”. Where in the GitHub repository can I find that dataset and what code was used to generate that dataset?

- Can you give an concrete example of a prompt that elicits a causal response from the LLM? For example a prompt in the medical domain that generates an answer to a question about ATE or a question about a CG ?

At the moment I am reading your paper and playing with the code, but the above questions still remain …

I am very curious !

**Comment by Kairong-Han** [comment]
Thank you for your letter. I will update "read me.md" in the github to show how to run my code. I hope this work may give your some help.



---- Replied Message ----
| From | Jan ***@***.***> |
| Date | 08/15/2024 22:06 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [Kairong-Han/Causal_Agent] Some questions on how to use the code from Causal Agent (Issue #1) |

Very interesting application of LLMs in the domain of Causal Inference!

I would like to replicate your results using your code, (not using GPT4 as LLM, but one or more of the open LLMs from GROQ, see for example https://github.com/groq/groq-api-cookbook).

My questions :

Can you tell me the exact steps I have to do using your code? It is not clear to me what to run in which order. Especially which output files have to be generated by which code in which order to do this.

Also I would like to know the process steps at a high level. Can you summarize them for me?

Where in the code are the numerical data files generated?

I am very curious !

**Comment by jbdatascience** [comment]
> Thank you for your letter. I will update "read me.md" in the github to show how to run my code. I hope this work may give your some help.
> […](#)
> ---- Replied Message ---- | From | Jan ***@***.***> | | Date | 08/15/2024 22:06 | | To | ***@***.***> | | Cc | ***@***.***> | | Subject | [Kairong-Han/Causal_Agent] Some questions on how to use the code from Causal Agent (Issue #1) | Very interesting application of LLMs in the domain of Causal Inference! I would like to replicate your results using your code, (not using GPT4 as LLM, but one or more of the open LLMs from GROQ, see for example https://github.com/groq/groq-api-cookbook). My questions : Can you tell me the exact steps I have to do using your code? It is not clear to me what to run in which order. Especially which output files have to be generated by which code in which order to do this. Also I would like to know the process steps at a high level. Can you summarize them for me? Where in the code are the numerical data files generated? I am very curious ! — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Thank you for your swift reply!
I will try if I can successfully run your examples in the readme file now (using another LLM).

Question: does the LLM used have to be capable of function calling / tool use?

**Comment by Kairong-Han** [comment]
LLM invoke tools（call function） by pass json format string as parameter to python code. This process is implemented by langchain framework. So it's no requirements for LLM to have tool use ability itself.
